.pre-commit-config.yaml

```diff
@@ -0,0 +1,193 @@
+        name: Add license for all Other files
+        exclude: ^\.github/.*$
+        files: >
+          \.conf$|^Makefile$|\.properties$
+        args:
+          - --comment-style
+          - "|#|"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
+      - id: insert-license
+        name: Add license for all Perl files
+        exclude: ^\.github/.*$
+        files: \.t$|\.pm$
+        args:
+          - --comment-style
+          - "|#|"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
+      - id: insert-license
+        name: Add license for all Python files
+        exclude: ^\.github/.*$
+        types: [python]
+        args:
+          - --comment-style
+          - "|#|"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
+      - id: insert-license
+        name: Add license for all Shell files
+        exclude: ^\.github/.*$
+        files: \.sh$
+        args:
+          - --comment-style
+          - "|#|"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
+      # - id: insert-license
+      #  name: Add license for all XML files
+      #  exclude: ^\.github/.*$
+      #  files: \.xml$
+      #  args:
+      #    - --comment-style
+      #    - "<!--||-->"
+      #    - --license-filepath
+      #    - license-templates/LICENSE.txt
+      #    - --fuzzy-match-generates-todo
+      - id: insert-license
+        name: Add license for all YAML files
+        exclude: ^\.github/.*$
+        types: [yaml]
+        files: \.(yaml|yml)$
+        args:
+          - --comment-style
+          - "|#|"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+      - id: check-case-conflict
+      - id: check-json
+      - id: check-merge-conflict
+      - id: check-xml
+      - id: check-yaml
+      - id: end-of-file-fixer
+        exclude: ^license-templates/LICENSE\.txt$
+      - id: fix-byte-order-marker
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+        exclude: ^t/node/remote-addr-ipv6\.t$
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.0.0
+    hooks:
+      - id: codespell
+        name: Run codespell
+        description: Check Spelling with codespell
+        entry: codespell --ignore-words=codespell.txt
+        exclude: ^docs/ar|^docs/es|^docs/zh|^.git|^.*\.crt$|^.*\.sum$
+  - repo: https://github.com/Calinou/pre-commit-luacheck
+    rev: v1.0.0
+    hooks:
+      - id: luacheck
+        name: Run luacheck
+        description: Checks Lua files with luacheck
+        exclude: ^apisix|^t/lib|^example/apisix/plugins/3rd-party\.lua$|^benchmark/fake-apisix/lua/apisix\.lua$
```

Comment:
Why exclude?

Reply:
`luacheck` seems to fail on `pre-commit` if you include these files